pytorch_lightning/core/datamodule.py

```diff
@@ -366,3 +362,63 @@ def get_init_arguments_and_types(cls) -> List[Tuple[str, Tuple, Any]]:
             name_type_default.append((arg, arg_types, arg_default))
 
         return name_type_default
+
+    @classmethod
+    def from_datasets(
+        cls,
+        train_dataset: Optional[Dataset] = None,
+        val_dataset: Optional[Union[Dataset, List[Dataset]]] = None,
+        test_dataset: Optional[Union[Dataset, List[Dataset]]] = None,
+        batch_size: int = 1,
+        num_workers: Optional[int] = None,
+    ):
+        r"""
+        Create an instance from torch.utils.data.Dataset.
+
+        Args:
+            train_dataset: (optional) Dataset to be used for train_dataloader()
+            val_dataset: (optional) Dataset or list of Dataset to be used for val_dataloader()
+            test_dataset: (optional) Dataset or list of Dataset to be used for test_dataloader()
+            batch_size: Batch size to use for each dataloader. Default is 1.
+            num_workers: Number of workers to use for each dataloader, if None, this will be set to the
+                number of CPUs available.
+
+        """
+        if num_workers is None:
+            num_workers = os.cpu_count()
+
+        if num_workers is None:
+            # TODO Should warn?
+            # warnings.warn("could not infer cpu count automatically, setting it to zero")
+            num_workers = 0
```

Comment:
just curious, in what case does `os.cpu_count()` returns `None`?

Reply:
If its "undetermined" https://docs.python.org/3/library/os.html#os.cpu_count, not sure exactly the case but I figure its good to check